src/orchestrator-process/src/lib.rs

```diff
@@ -716,6 +716,14 @@ async fn spawn_process(
 
     impl Drop for KillOnDropChild {
         fn drop(&mut self) {
+            if let Some(pid) = self.0.id().and_then(|id| i32::try_from(id).ok()) {
+                let _ = nix::sys::signal::kill(
```

Comment:
I'm kind of conflicted about this. I consider it a feature that the process orchestrator SIGKILLs processes when they're done. I wouldn't want someone adding an "onexit" handler to `clusterd` for example. If it's easier, might be nice to only run this logic when a process wrapper is in use.